pkg/rpc/client/client_test.go

```diff
@@ -284,3 +284,36 @@ func TestScanner_Scan(t *testing.T) {
 		})
 	}
 }
+
+func TestScanner_Insecure(t *testing.T) {
+	type args struct {
+		remoteUrl string
+		insecure  bool
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr string
+	}{
+		{
+			name: "happy path: insecure",
+			args: args{
+				remoteUrl: "https://www.example.com",
+				insecure:  true,
+			},
+		},
+		{
+			name: "happy path: secure",
+			args: args{
+				remoteUrl: "https://www.example.com",
+				insecure:  false,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewProtobufClient(RemoteURL(tt.args.remoteUrl), Insecure(tt.args.insecure))
```

Comment:
Looks like these cases don't test anything since the error is always nil.

Reply:
Re-wrote the tests in the latest commits to connect against an insecure TLS server.